rcl/src/rcl/security.c

```diff
@@ -132,58 +132,67 @@ char * rcl_get_secure_root(
   const char * name,
   const rcl_allocator_t * allocator)
 {
-  bool ros_secure_directory_override = true;
+  bool ros_secure_enclave_override = true;
 
   // find out if either of the configuration environment variables are set
   const char * env_buf = NULL;
   if (NULL == name) {
     return NULL;
   }
 
-  if (rcutils_get_env(ROS_SECURITY_DIRECTORY_OVERRIDE, &env_buf)) {
+  // check if enclave override environment variable is empty
+  if (rcutils_get_env(ROS_SECURITY_ENCLAVE_OVERRIDE, &env_buf)) {
     return NULL;
   }
   if (!env_buf) {
     return NULL;
   }
   if (0 == strcmp("", env_buf)) {
-    // check root directory if override directory environment variable is empty
-    if (rcutils_get_env(ROS_SECURITY_ROOT_DIRECTORY_VAR_NAME, &env_buf)) {
-      return NULL;
-    }
-    if (!env_buf) {
-      return NULL;
-    }
-    if (0 == strcmp("", env_buf)) {
-      return NULL;  // environment variable was empty
-    }
-    ros_secure_directory_override = false;
+    ros_secure_enclave_override = false;
   }
+  char * ros_secure_enclave_override_env = rcutils_strdup(env_buf, *allocator);
 
-  // found a usable environment variable, copy into our memory before overwriting with next lookup
-  char * ros_secure_root_env = rcutils_strdup(env_buf, *allocator);
+  // check if keystore environment variable is empty
+  if (rcutils_get_env(ROS_SECURITY_KEYSTORE_VAR_NAME, &env_buf)) {
+    return NULL;
+  }
```

Comment:
`rcutils_get_env()` returns non-NULL if an error was encountered (in which case it returns an error string). Would it be worth showing that error to the user?

Reply:
I was just keeping with the existing return NULL after any error. Are there any existing patterns else where in RCL that do more transparent error handling that we could do in a follow up PR?

Reply:
you can do:
```c
char * error_str = rcutils_get_env(ROS_SECURITY_KEYSTORE_VAR_NAME, &env_buf);
if (error_str) {
  RCUTILS_LOG_ERROR("rcutils_get_env failed: %s\n", error_str);
  return NULL;
}
```

Reply:
See 1ee7094